plugins/FakeHSIEventGenerator.cpp

```diff
@@ -289,6 +283,36 @@ FakeHSIEventGenerator::do_hsievent_work(std::atomic<bool>& running_flag)
     } else {
       continue;
     }
+
+    // sleep for the configured event period, if trigger ticks are not 0, otherwise do not send anything
+    if (m_active_trigger_rate.load() > 0)
+    {
+      auto prev_gen_time = std::chrono::steady_clock::now();
+      auto next_gen_time = prev_gen_time + std::chrono::microseconds(m_event_period.load());
```

Comment:
How strict is m_event_period? I'm just wondering if it would be better to base next_gen_time off of a steady_clock and  a counter, e.g.
```CPP
auto next_gen_time = run_start_time + std::chrono::microseconds(counter * m_event_period.load());
```

Reply:
I guess a similar effect could be obtained by declaring prev_gen_time outside the loop and updating it with the previous next_gen_time

Reply:
Thanks Eric. Ideally, `m_event_period` would always correspond to a particular number of ticks, however currently this is not always the case due to the way the `TimestampEstimator` works. In any case, I'll have a look at putting in the suggested changes today.

Reply:
@eflumerf I have implemented the suggested change.